tools/autograd/templates/Functions.cpp

```diff
@@ -902,6 +902,40 @@ Tensor log_softmax_double_backward(const Tensor & grad, const Tensor & grad_outp
   return z * grad_output.sum(dim, true) * ((grad * z).sum(dim, true) - grad);
 }
 
+Tensor binary_cross_entropy_double_backward(const Tensor & grad_output, const Tensor & grad, const Tensor & input, const Tensor & target, const Tensor& weight, int64_t reduction) {
+  auto eps = 1e-12;
+  auto num = input * input - 2 * input * target + target;
+  auto denom = (input + eps) * (input + eps) * (1 - input + eps) * (1 - input + eps);
+  auto output = num / denom;
+  output *= (grad * grad_output);
```

Comment:
I think this would be better as: `output.mul_(grad).mul_(grad_output);`.
That way you do not create an intermediary Tensor of the size of the output.